File: java/src/main/java/edu/unialfa/java/model/Questao.java
package edu.unialfa.java.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "questoes")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Questao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "prova_id", nullable = false)
    @JsonBackReference
    private Prova prova;

    @Column(columnDefinition = "TEXT")
    private String enunciado;

    private String alternativaA;

    private String alternativaB;

    private String alternativaC;

    private String alternativaD;

    private String alternativaE;

    @Column(length = 1)
    private String respostaCorreta; // A, B, C, D ou E

    private Double valor;
}

File: correcaogabarito_app/lib/models/questao.dart
class Questao {
  final int id;
  final String enunciado;
  final String alternativaA;
  final String alternativaB;
  final String alternativaC;
  final String alternativaD;
  final String alternativaE;
  final String respostaCorreta;
  final double valor;

  Questao({
    required this.id,
    required this.enunciado,
    required this.alternativaA,
    required this.alternativaB,
    required this.alternativaC,
    required this.alternativaD,
    required this.alternativaE,
    required this.respostaCorreta,
    required this.valor,
  });

  factory Questao.fromJson(Map<String, dynamic> json) {
    return Questao(
      id: json['id'],
      enunciado: json['enunciado'],
      alternativaA: json['alternativaA'],
      alternativaB: json['alternativaB'],
      alternativaC: json['alternativaC'],
      alternativaD: json['alternativaD'],
      alternativaE: json['alternativaE'],
      respostaCorreta: json['respostaCorreta'],
      valor: (json['valor'] as num).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'enunciado': enunciado,
      'alternativaA': alternativaA,
      'alternativaB': alternativaB,
      'alternativaC': alternativaC,
      'alternativaD': alternativaD,
      'alternativaE': alternativaE,
      'respostaCorreta': respostaCorreta,
      'valor': valor,
    };
  }
}
